dotnet/src/Experimental/Assistants/IAssistant.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Experimental.Assistants;
+
+/// <summary>
+/// Represents an assistant that can call the model and use tools.
+/// </summary>
+public interface IAssistant
```

Comment:
Do we expect customers to implement their own Assistants?

We are favouring public classes because once we ship and API we cannot change it. Not an issue now but requires thought later.

Reply:
This allows for mocking in the callers application pattern.